internal/daemon/worker/worker.go

```diff
@@ -188,26 +188,81 @@ func (w *Worker) Start() error {
 		return fmt.Errorf("error starting worker listeners: %w", err)
 	}
 
-	nodeCreds, err := types.NewNodeCredentials(w.baseContext, w.WorkerAuthStorage, nodeenrollment.WithWrapper(w.conf.WorkerAuthStorageKms))
-	if err != nil {
-		return fmt.Errorf("error generating new node creds: %w", err)
-	}
+	// Check for worker auth creds and if not create new ones
+	var createNodeAuthCreds bool
+	var createFetchRequest bool
+	nodeCreds, err := types.LoadNodeCredentials(w.baseContext, w.WorkerAuthStorage, nodeenrollment.CurrentId, nodeenrollment.WithWrapper(w.conf.WorkerAuthStorageKms))
+	switch {
+	case err == nil:
+		if nodeCreds == nil {
+			event.WriteSysEvent(w.baseContext, op, "no error loading worker auth creds but nil creds, creating new creds for registration")
+			createNodeAuthCreds = true
+			createFetchRequest = true
+			break
+		}
 
-	req, err := nodeCreds.CreateFetchNodeCredentialsRequest(w.baseContext)
-	if err != nil {
-		return fmt.Errorf("error creating fetch credentials request: %w", err)
-	}
-	reqBytes, err := proto.Marshal(req)
-	if err != nil {
-		return fmt.Errorf("error marshaling fetch credentials request: %w", err)
+		// Check that we have valid creds, or that we have generated creds but
+		// simply are still waiting on authentication (in which case we don't
+		// want to invalidate what we've already sent)
+		var validCreds bool
+		switch len(nodeCreds.CertificateBundles) {
+		case 0:
+			// Still waiting on initial creds, so don't invalidate the request
+			// by creating new credentials. However, we will generate and
+			// display a new valid request in case the first was lost.
+			createFetchRequest = true
+
+		default:
+			now := time.Now()
+			for _, bundle := range nodeCreds.CertificateBundles {
+				if bundle.CertificateNotBefore.AsTime().Before(now) && bundle.CertificateNotAfter.AsTime().After(now) {
+					validCreds = true
+					break
+				}
+			}
+
+			// Certificates are both expired, create new credential request
+			createNodeAuthCreds = !validCreds
+			createFetchRequest = !validCreds
+		}
+
+	case errors.Is(err, nodeenrollment.ErrNotFound):
+		createNodeAuthCreds = true
+		createFetchRequest = true
+
+	default:
+		// Some other type of error happened, bail out
+		return fmt.Errorf("error loading worker auth creds: %w", err)
 	}
-	w.WorkerAuthRegistrationRequest = base58.FastBase58Encoding(reqBytes)
-	if err != nil {
-		return fmt.Errorf("error encoding registration request: %w", err)
+
+	if createNodeAuthCreds {
```

Comment:
mind adding a comment that this MUST happen before `if createFetchRequest`... it's great you have a test for that conditional as well btw.

Reply:
Done!